src/tapedfunction.jl

```diff
@@ -179,20 +176,29 @@ function Base.show(io::IO, instr::GotoInstruction)
     println(io, "GotoInstruction(", instr.condition, ", dest=", instr.dest, ")")
 end
 
-function (instr::Instruction{F})(tf::TapedFunction) where F
-    # catch run-time exceptions / errors.
-    try
-        func = val(_lookup(tf, instr.func))
-        inputs = map(x -> val(_lookup(tf, x)), instr.input)
-        output = func(inputs...)
-        _update_var!(tf, instr.output, output)
-        tf.counter += 1
-    catch e
-        println("counter=", tf.counter)
-        println("tf=", tf)
-        println(e, catch_backtrace());
-        rethrow(e);
+@generated function (instr::Instruction{F, N})(tf::TapedFunction) where {F, N}
+    arity = instr.parameters[2]
+    body = quote
+        try
+            func = val(_lookup(tf, instr.func))
+            output = func() # will inject arguments later
+            _update_var!(tf, instr.output, output)
+            tf.counter += 1
+        catch e
+            # catch run-time exceptions / errors.
+            println("counter=", tf.counter)
+            println("tf=", tf)
+            println(e, catch_backtrace());
+            rethrow(e);
+        end
     end
+    body = MacroTools.striplines(body)
```

Comment:
Is this the only place where MacroTools is used? This line could be replaced with

```suggestion
    Base.remove_linenums!(body)
```

Reply:
Thank you, updated.